src/components/SelectionPopUpComponent.js

```diff
@@ -54,14 +65,50 @@ class SelectionPopUpComponent {
    * @param position last mouse action position
    */
   #handleSelection = (position) => {
+    let component = this.components;
     const selectedText = getCurrentDOMSelection().trim();
     if (selectedText !== "") {
-      //TODO:- PopUp here -> On button dispatch the event to display summary
+      component.popupSelectedText.classList.remove("hidden");
+      component.popupSelectedText.style.top = position.y + "px";
+      component.popupSelectedText.style.left = position.x + 20 + "px";
       EventHandlerService.publish(
         PDFLEvents.onTextSelectionReady,
         selectedText
       );
     }
+
+    /* Hide the popup events */
+    const hidePopupTimeout = setTimeout(() => {
+      this.hidePopup();
+    }, POPUP_DISAPPEAR_TIMEOUT);
+    component.popupSelectedText.addEventListener("mouseenter", (event) => {
```

Comment:
This event is added everytime you select text? If it overrides the existing one then it's fine.